test/test_encodings.rb

```diff
@@ -0,0 +1,59 @@
+require File.dirname(__FILE__) + '/helper'
+
+class TestEncodings < Test::Unit::TestCase
+  def setup
+    @internal_encoding = Encoding.default_internal
+    @dir = Dir.mktmpdir("grit-test")
+    `cd "#{@dir}" && git  init`
+    @r = Repo.new(@dir)
+  end
+
+  def teardown
+    Encoding.default_internal = @internal_encoding
+    FileUtils.rmdir @dir
+  end
+
+  def test_english_commit_message_encoding
+    index = @r.index
+    index.add("README.md", "English text")
+    sha = index.commit("English commit message")
+
+    commit = @r.commit(sha)
+
+    assert_equal Encoding::UTF_8, commit.message.encoding
+  end
+
+  def test_japanese_commit_message_encoding
+    index = @r.index
+    index.add("README.md", "ャストスケジ可能なマッチングも")
+    sha = index.commit("ャストスケジ")
+
+    Encoding.default_internal = Encoding::UTF_8
+    commit = @r.commit(sha)
+
+    assert_equal Encoding::UTF_8, commit.message.encoding
+  end
+
+  def test_english_file_encoding
+    index = @r.index
+    index.add("README.md", "English text")
+    sha = index.commit("English commit message")
+
+    blob = @r.commit(sha).tree / "README.md"
+    file_contents = blob.data
+
+    assert_equal Encoding::UTF_8, file_contents.encoding
+  end
+
+  def test_japanese_file_encoding
```

Comment:
I just did full equality on the expected strings:

https://github.com/brynary/grit/commit/6e11f9907a7cbb385f035156ccfb2284c86bbc57#diff-e93b01431b04f009beb74f98e4341559R35
https://github.com/brynary/grit/commit/6e11f9907a7cbb385f035156ccfb2284c86bbc57#diff-e93b01431b04f009beb74f98e4341559R59